Security vulnerability CVE-2023-47248 is already fixed upstream since Arrow 14.0.1
see https://github.com/pitrou/pyarrow-hotfix?tab=readme-ov-file#description
author: Alex Domingo (Vrije Universiteit Brussel)
--- PKG-INFO.orig	2024-01-31 10:24:17.114114000 +0100
+++ PKG-INFO	2024-01-31 10:22:43.428088000 +0100
@@ -8,7 +8,6 @@
 Classifier: Programming Language :: Python :: 3.11
 Classifier: Programming Language :: Python :: 3.12
 Requires-Dist: pyarrow >=8
-Requires-Dist: pyarrow-hotfix
 Requires-Dist: pandas ; extra == 'pandas'
 Requires-Dist: mypy ; extra == 'devel'
 Requires-Dist: ruff >=0.1.5 ; extra == 'devel'
--- pyproject.toml.orig	2024-01-31 10:24:37.004732357 +0100
+++ pyproject.toml	2024-01-31 01:13:56.729671621 +0100
@@ -19,7 +19,6 @@
 ]
 dependencies = [
     "pyarrow>=8",
-    "pyarrow-hotfix",    
 ]
 
 [project.optional-dependencies]
--- deltalake/table.py.orig	2024-01-31 10:25:17.340707000 +0100
+++ deltalake/table.py	2024-01-31 10:22:13.865436632 +0100
@@ -25,7 +25,6 @@
 import pyarrow
 import pyarrow.dataset as ds
 import pyarrow.fs as pa_fs
-import pyarrow_hotfix  # noqa: F401; addresses CVE-2023-47248; # type: ignore
 from pyarrow.dataset import (
     Expression,
     FileSystemDataset,
--- python/deltalake/table.py.orig	2024-01-31 10:25:28.723000000 +0100
+++ python/deltalake/table.py	2024-01-31 10:22:00.855595977 +0100
@@ -25,7 +25,6 @@
 import pyarrow
 import pyarrow.dataset as ds
 import pyarrow.fs as pa_fs
-import pyarrow_hotfix  # noqa: F401; addresses CVE-2023-47248; # type: ignore
 from pyarrow.dataset import (
     Expression,
     FileSystemDataset,
